data/binding/json.go

```diff
@@ -0,0 +1,329 @@
+package binding
+
+import (
+	"errors"
+	"sync"
+
+	"fyne.io/fyne/v2/data/binding"
+
+	jsonvalue "github.com/Andrew-M-C/go.jsonvalue"
+)
+
+// JSONValue supports binding a jsonvalue.V{}
+type JSONValue interface {
+	binding.DataItem
+	sync.Locker
+	RLock()
+	RUnlock()
+
+	Get() (*jsonvalue.V, error)
+	Set(*jsonvalue.V) error
+	GetItemString(firstParam interface{}, params ...interface{}) (binding.String, error)
+	GetItemFloat(firstParam interface{}, params ...interface{}) (binding.Float, error)
+	GetItemBool(firstParam interface{}, params ...interface{}) (binding.Bool, error)
+}
+
+type databoundJSON struct {
+	JSONValue
+
+	self   binding.Untyped
+	lock   sync.RWMutex
+	source binding.String
+	last   string
+	err    error
+}
+
+// Internal type for the children accessor
+type childJSON struct {
+	source JSONValue
+	err    error
+
+	target []interface{}
+	first  interface{}
+}
+
+type childJSONString struct {
+	binding.String
+	generic childJSON
+}
+
+type childJSONFloat struct {
+	binding.Float
+	generic childJSON
+}
+
+type childJSONBool struct {
+	binding.Bool
+	generic childJSON
+}
+
+var (
+	errWrongType = errors.New("wrong type provided")
+)
+
+// NewJSONFromDataString return a data binding to a `jsonvalue.V{}` synchronized with the `String`
+// binding used to create the new binding.
+func NewJSONFromDataString(data binding.String) (JSONValue, error) {
```

Comment:
I wonder if the word `Data` in here is not needed? The Input is a `String`

Reply:
I think that's a good suggestion, I have dropped Data from the name.